### ISSUE TITLE
Log when a connection to the database is closed

### ISSUE THREAD

**Issue by janpio** [prisma/prisma#3993]
Currently the Query Engine (via Prisma Client) only logs the creation of the connection pool after startup and when a connection is created (vs. taken as existing from the connection pool) - but not when it is closed. This makes it hard to observe the behavior of the Query Engine and the connection pool.

We should log every time a connection is closed.

**Comment by dpetrick** [comment]
If @pimeys has capacity at some point, I would like to pair on this issue.

**Comment by pimeys** [comment]
This would be a bit easier if we define the conditions. We do not close connections, but connections can get stale and dropped when they start giving errors. Logging a dropped connection should then happen in the connection pool crate (mobc) or even deeper in the database crates.

**Comment by janpio** [comment]
`$disconnect` does not close a connection?

**Comment by pimeys** [comment]
Yeah, it tears down the whole engine, so naturally we also disconnect. Implementing some logging _might_ or might not include adding the feature to the database crates... And then of course logging it there.
